### DISCUSSION TITLE
How to restrict the list of enabled feature toggles from Proxy by Projects?

### DISCUSSION THREAD

**Discussion by gsannec** [Unleash/unleash#5782] [Q&A] [answered]
Hi!

When requesting the list of feature toggles from the Proxy API, we get all feature toggles from all of the Projects.

If there are 50 projects and 50 enabled toggles per project then one small front-end app, that uses a few toggles and connects via Proxy, will get 2500 toggles. 

Is it possible to restrict the list of enabled feature toggles from Proxy by projects?

For example, I've got a front-end app AAA and want to get toggles for it from Project A only, and I've got another front-end app BBB and want to get toggles for it from Project B only.

**Comment by chriswk** [comment, marked as answer]
Hi @gsannec - This should be doable using API keys, they can be limited by projects as well as environments. See https://docs.getunleash.io/reference/api-tokens-and-client-keys#proxy-client-keys

**Comment by gsannec** [comment]
That's interesting!
We use unleash-api 4.15.4 (we don't have client-side tokens) and unleash-proxy 0.10.4.
I tryed to use server-side token with access to only one project as a client key for application, but it looks like Proxy uses this parameter only for authentication, so I received toggles from all projects again.
